File: packages/csharp/HtmlToMarkdown/Metadata/TableData.cs
using System.Collections.Generic;
using System.Text.Json.Serialization;

namespace HtmlToMarkdown.Metadata;

/// <summary>
/// A single table extracted during HTML to Markdown conversion.
/// </summary>
public record TableData
{
    /// <summary>
    /// 2D list of cell string values, in row-major order.
    /// </summary>
    [JsonPropertyName("cells")]
    public List<List<string>> Cells { get; init; } = [];

    /// <summary>
    /// The rendered Markdown representation of the table.
    /// </summary>
    [JsonPropertyName("markdown")]
    public string Markdown { get; init; } = string.Empty;

    /// <summary>
    /// Boolean list indicating which rows are header rows.
    /// </summary>
    [JsonPropertyName("is_header_row")]
    public List<bool> IsHeaderRow { get; init; } = [];
}

File: packages/java/src/main/java/dev/kreuzberg/htmltomarkdown/TableData.java
package dev.kreuzberg.htmltomarkdown;

import com.fasterxml.jackson.annotation.JsonProperty;
import java.util.List;

/**
 * A single table extracted during HTML to Markdown conversion.
 *
 * @param cells 2D list of cell string values, row-major order
 * @param markdown the rendered Markdown representation of the table
 * @param isHeaderRow boolean list indicating which rows are header rows
 */
public record TableData(
    @JsonProperty("cells") List<List<String>> cells,
    @JsonProperty("markdown") String markdown,
    @JsonProperty("is_header_row") List<Boolean> isHeaderRow) {}
